src/bson.ts

```diff
@@ -216,7 +216,7 @@ export function deserialize(
   buffer: Buffer | ArrayBufferView | ArrayBuffer,
   options: DeserializeOptions = {}
 ): Document {
-  return internalDeserialize(ensureBuffer(buffer), options);
+  return internalDeserialize(buffer instanceof Buffer ? buffer : ensureBuffer(buffer), options);
```

Comment:
We have an `isBuffer` method in utils, I think its check is faster and preferred for checking the type here.

Reply:
capturing from meeting: `instanceof` is preferred